svir/dialogs/viewer_dock.py

```diff
@@ -262,14 +265,113 @@ def create_fields_multiselect(self):
             self.fields_multiselect, self.iface.activeLayer())
 
     def create_stats_multiselect(self):
-        title = 'Select statistics to plot'
+        title = 'Select statistics'
         self.stats_multiselect = ListMultiSelectWidget(title=title)
         self.stats_multiselect.setSizePolicy(
             QSizePolicy.MinimumExpanding, QSizePolicy.MinimumExpanding)
         self.typeDepVLayout.addWidget(self.stats_multiselect)
         self.stats_multiselect.selection_changed.connect(
             self.refresh_feature_selection)
 
+    def create_tag_names_multiselect(self):
+        title = 'Select tag names'
+        self.tag_names_multiselect = ListMultiSelectWidget(title=title)
+        self.tag_names_multiselect.setSizePolicy(
+            QSizePolicy.MinimumExpanding, QSizePolicy.MinimumExpanding)
+        self.tag_names_multiselect.selected_widget.setSelectionMode(
+            QAbstractItemView.SingleSelection)
+        self.tag_names_multiselect.unselected_widget.setSelectionMode(
+            QAbstractItemView.SingleSelection)
+        self.tag_names_multiselect.select_all_btn.hide()
+        self.tag_names_multiselect.deselect_all_btn.hide()
+        self.typeDepVLayout.addWidget(self.tag_names_multiselect)
+        self.tag_names_multiselect.unselected_widget.itemClicked.connect(
+            self.populate_tag_values_multiselect)
+        self.tag_names_multiselect.selected_widget.itemClicked.connect(
+            self.populate_tag_values_multiselect)
+        self.tag_names_multiselect.unselected_widget.itemDoubleClicked.connect(
+            self.populate_tag_values_multiselect)
+        self.tag_names_multiselect.selected_widget.itemDoubleClicked.connect(
+            self.populate_tag_values_multiselect)
+        self.tag_names_multiselect.selection_changed.connect(
+            self.update_selected_tag_names)
+
+    def create_tag_values_multiselect(self):
+        title = 'Select tag values'
+        self.tag_values_multiselect = ListMultiSelectMonoWidget(
+            message_bar=self.iface.messageBar(), title=title)
+        self.tag_values_multiselect.setSizePolicy(
+            QSizePolicy.MinimumExpanding, QSizePolicy.MinimumExpanding)
+        self.tag_values_multiselect.select_all_btn.hide()
+        self.tag_values_multiselect.deselect_all_btn.hide()
+        self.typeDepVLayout.addWidget(self.tag_values_multiselect)
+        self.tag_values_multiselect.selection_changed.connect(
+            self.update_selected_tag_values)
+
+    def populate_tag_values_multiselect(self, item):
+        tag_name = item.text()
+        self.current_tag_name = tag_name
+        self.tag_values_multiselect.setTitle(
+            'Select values for tag: %s' % tag_name)
+        selected_tag_values = [
```

Comment:
You may want to give a name to `self.tags[tag_name]['values']`

Reply:
Done